### PR TITLE
adds @graph-paper/chip

### LINKED ISSUE #31: @graph-paper/chip
We need a chip or tag style component for a few internal tools. This issue will track the developer needs around this:

- need a `Chip` component that adheres to our style system
- need `Chip` to be _togglable_ by prop – a `togglable` flag + `toggled={true}` should change to some default styling.
- need the `Chip` to be _cancellable_ by prop – a `cancelable` flag that puts a small x button + an `on:cancel` event. The cancel event should transition locally with `fold` (like slide but horizontal).

### PR DESCRIPTION
**Pull request by hamilton**
Closes #31. This is a needed component for the GUD revamp, so figured I'd port over and clean up what I have already.

graph-paper-chips-alpha

- [x] basic `Chip` / `ChipSet` component
- [x] add `filterable` functionality / styling
- [x] add `cancelable` functionality / styling
- [x] add `README.md`
- [x] update `CHANGELOG.md`

### CODE REVIEW

**Comment by hamilton** [comment]
@wlach @openjck not sure if either of you want to review this particular PR when you have the time. I'm currently blocked on a part of GUD pending this merge / publish. Alternatively, I'm happy to do my own review / merge if that works for y'all.

File: packages/chip/Chip.svelte [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  const marginRight = parseFloat(style.marginBottom);
+  const borderLeftWidth = parseFloat(style.borderLeftWidth);
+  const borderRightWidth = parseFloat(style.borderRightWidth);
+
+  return {
+    delay,
+    duration,
+    easing,
+    css: (t) => 'overflow: hidden;'
+      + 'white-space: nowrap;'
```
Comment (wlach) [inline packages/chip/Chip.svelte:41]: Are the `+`'s necessary?

File: packages/chip/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Chip
+
+A chip is a discrete interaction element conveying information that is filterable, cancelable, and clickable. More information behind chips can be found in the [Material Design documentation](https://material.io/components/chips/#input-chips).
```
Comment (wlach) [inline packages/chip/README.md:3]:
```suggestion
A chip is a discrete interaction element conveying information that is filterable, cancelable, and clickable. More information about chips can be found in the [Material Design documentation](https://material.io/components/chips/#input-chips).
```

File: packages/icons/Cancel.svelte [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (wlach) [inline packages/icons/Cancel.svelte:1]: I don't think it's necessary to put this license boilerplate in here, a link back to the Photon Design System should be sufficient (see suggestion below).

File: packages/icons/Cancel.svelte [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+   <!-- This icon was adapted from the Photon Design System. -->
```
Comment (wlach) [inline packages/icons/Cancel.svelte:4]:
```suggestion
   <!-- This icon was adapted from the Photon Design System (licensed under the MPL v2.0): https://design.firefox.com/photon/ -->
```

File: packages/chip/stories/Chip01.svelte [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+addCancelable();
+addCancelable();
+addCancelable();
+
+let multipleFilters = [0];
+
+function handleMultipleFilter(v) {
+  return () => {
+    if (multipleFilters.includes(v)) multipleFilters = [...multipleFilters.filter((di) => di !== v)];
+    else multipleFilters = [...multipleFilters, v];
```
Comment (wlach) [inline packages/chip/stories/Chip01.svelte:41]: Could do this in a one-liner, which I think might be a bit clearer:

```js
multipleFilters = [...multipleFilters.filter((di) => di !== v)].concat(multipleFilters.includes(v) ? [v], []);

File: packages/chip/Chip.svelte (hunk 1 above)
Comment (hamilton) [reply packages/chip/Chip.svelte:41]: This was adapted from Svelte's native `slide` transition. I can turn this into a string template though :thumbsup:

File: packages/icons/Cancel.svelte (hunk 3 above)
Comment (hamilton) [reply packages/icons/Cancel.svelte:1]: Got it, we should probably have a conversation about this in greater detail. I left in the license boilerplate on everything in `package/icons`. Happy to remove it across the board in that directory in this PR.

File: packages/icons/Cancel.svelte (hunk 4 above)
Comment (hamilton) [reply packages/icons/Cancel.svelte:4]: 👍 great idea

**Pull request merged**